### PR TITLE
Updated clusterroles

### PR DESCRIPTION
**Pull request by ghanshyam-11**
Updated clusterroles for sidecars and operator

## Pull request checklist

```release-notes

```

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature Enhancement
- [ ] Test Automation 
- [ ] Code Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Community Operator listing 
- [ ] Other (please describe): 

## What is the current behavior?

-

## What is the new behavior?

-

## How risky is this change?
- [ ] Small, isolated change
- [x] Medium, requires regression testing
- [ ] Large, requires functional and regression testing

### CODE REVIEW

**Comment by ghanshyam-11** [comment]
1) "*" indicates the verbs: [get, list, watch, delete, create, update, patch, deletecollection]
   ref: [](https://kubernetes.io/docs/reference/access-authn-authz/authorization/#determine-the-request-verb)
   The "deletecollection" verb is unnecessary, hence removed for some resources like deployments, Daemonsets, Statefulsets, 
   replicasets, clusterrole, clusterrolebindings.

2) In Snapshotter clusterrole, For volumesnapshotcontents.snapshot.storage.k8s.io/status resource, only "patch" verb should 
    be sufficient but in the IBM CSI code, its ["patch", 'update'], hence removed 'update'.

    ref : [](https://github.com/kubernetes-csi/external-snapshotter/blob/master/deploy/kubernetes/csi-snapshotter/rbac-csi-snapshotter.yaml)

3) In Resizer Clusterrole, for Resource: persistentvolumes
      has "[get list watch patch update]" verbs.
      But to expand/shrink the volume, only the volumesize field needs to be modified, hence "patch" looks sufficient.
      Therefore, "update" should be removed.

     ref: [](https://github.com/kubernetes-csi/external-resizer/blob/master/deploy/kubernetes/rbac.yaml)
           [](https://kubernetes-csi.github.io/docs/volume-expansion.html)

4) In Resizer Clusterrole, for persistentvolumeclaims/status
      IBM resizer clusterroles include ["patch", "update"] but the ref doc only includes "patch"\

   ref: []( https://github.com/kubernetes-csi/external-resizer/blob/master/deploy/kubernetes/rbac.yaml)

**Comment by ghanshyam-11** [comment]
Operator image containing above changes can be pulled using:
`docker pull quay.io/ghanshyam_sharma1/operator_repo:operator_v1.0.0
`

**Review by amdabhad — APPROVED** [review]

File: operator/controllers/internal/csiscaleoperator/csiscaleoperator_package.go [hunk 1]
```diff
@@ -369,7 +369,7 @@ func (c *CSIScaleOperator) GenerateSnapshotterClusterRole() *rbacv1.ClusterRole
 			{
 				APIGroups: []string{snapshotStorageApiGroup},
 				Resources: []string{volumeSnapshotContentsStatusResource},
-				Verbs:     []string{verbUpdate, verbPatch},
```
Comment (Jainbrt) [inline operator/controllers/internal/csiscaleoperator/csiscaleoperator_package.go:372]: @amdabhad @ghanshyam-11 are we sure that we are not breaking anything while remove verbUpdate operation ?

Comment (amdabhad) [reply operator/controllers/internal/csiscaleoperator/csiscaleoperator_package.go:372]: Thanks @Jainbrt ! @ghanshyam-11 even though our regression worked fine with these clusterroles and update verb had to be removed from resizer, looking at [community snapshotter clusterroles](https://github.com/kubernetes-csi/external-snapshotter/blob/master/deploy/kubernetes/csi-snapshotter/rbac-csi-snapshotter.yaml), it seems we need update verb here?

Comment (amdabhad) [reply operator/controllers/internal/csiscaleoperator/csiscaleoperator_package.go:372]: update verb is added back by https://github.com/IBM/ibm-spectrum-scale-csi/pull/811, community snapshotter issue for reference: https://github.com/kubernetes-csi/external-snapshotter/pull/707

**Pull request merged**